File: JavaScript/Comment-to-Code Ratio/workspace/comment_to_code_ratio_benchmark/cases/heavily_commented.js
/**
 * Heavily commented JavaScript module for Comment-to-Code Ratio validation.
 * Includes block and single-line comments throughout the file.
 */

/* Utility helpers documented with block comments. */

/**
 * Add two numbers and return the sum.
 * @param {number} left
 * @param {number} right
 * @returns {number}
 */
function add(left, right) {
  // Return the sum of both operands.
  return left + right;
}

/**
 * Multiply two numbers.
 */
function multiply(left, right) {
  // Compute product inline.
  return left * right;
}

/* Aggregate three values. */
function sumThree(a, b, c) {
  // Accumulate all values.
  return a + b + c;
}

module.exports = { add, multiply, sumThree };

File: C/Comment-to-Code Ratio/workspace/comment_to_code_ratio_benchmark/cases/heavily_commented.c
/* Heavily commented C source for Comment-to-Code Ratio validation. */

#include "heavily_commented.h"

/* Add two integers with inline documentation. */
int hc_add(int left, int right) {
    /* Return the sum of left and right operands. */
    return left + right;
}

/* Multiply two integers. */
int hc_multiply(int left, int right) {
    int product; /* Holds intermediate product value. */
    product = left * right;
    return product;
}

/* Aggregate values from a small fixed-size array. */
int hc_sum_three(int a, int b, int c) {
    /* Accumulate all three values. */
    return a + b + c;
}
